cvat/apps/engine/models.py

```diff
@@ -391,6 +422,33 @@ def get_tmp_dirname(self):
     def __str__(self):
         return self.name
 
+    def delete(self, using=None, keep_parents=False):
+        from cvat.apps.events.signals import resource_delete
+        from cvat.apps.events.utils import clear_cache
+        try:
```

Comment:
I will recommend filling the cache in pre_delete and flushing it in post_delete. Thus we can be sure that we generate events only after an object is deleted. 

1. pre_delete -> store all required information inside the cache (you can prepare ready to use event here)
2. post_delete -> check that you have the required information inside the cache and flush it, otherwise handle the signal as usual.